v2/internal/runner/runner.go

```diff
@@ -785,3 +789,31 @@ func (r *Runner) SaveResumeConfig(path string) error {
 
 	return os.WriteFile(path, data, os.ModePerm)
 }
+
+// replace $VAR_EXAMPLE with the correct variable in os ENV
+func assignEnvVarToReportingOpt(val reflect.Value) {
+	for i := 0; i < val.NumField(); i++ {
+		field := val.Field(i)
+		if field.Kind() == reflect.Ptr {
+			if field.IsNil() {
+				continue
+			}
+			field = field.Elem()
+		}
+		if field.Kind() == reflect.Struct {
+			assignEnvVarToReportingOpt(field)
+		} else {
+			if field.CanAddr() {
+				fieldType := val.Type().Field(i)
+				if fieldType.Tag.Get("yaml") != "" {
+					if field.Kind() == reflect.String {
+						valueStr := fmt.Sprintf("%v", field.Interface())
+						if strings.HasPrefix(valueStr, "$") {
```

Comment:
The implementation doesn't seem correct, every value starting with $ is replaced, even if the environment variable doesn't exist. If the environment variable doesn't exist, the field should be left intact